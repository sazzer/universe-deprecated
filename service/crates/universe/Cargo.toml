[package]
name = "universe"
version = "0.1.0"
authors = ["Graham Cox <graham@grahamcox.co.uk>"]
edition = "2018"

[dependencies]
config = "0.10.1"
dotenv = "0.15.0"
serde = {version = "1.0.104", features=["derive"] }
tracing = "0.1.12"
tracing-log = "0.1.1"
tracing-subscriber = "0.2.1"

universe_webapp = { path = "../webapp" }

[dev-dependencies]
