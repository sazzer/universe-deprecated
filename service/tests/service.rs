use insta::assert_snapshot;
use lazy_static::lazy_static;
use rocket::local::{Client, LocalResponse};
use std::fmt::Write;
use testcontainers::{clients::Cli, images::postgres::Postgres, Container, Docker};
use tracing::info;
use universe::Service;

lazy_static! {
    static ref DOCKER: Cli = { Cli::default() };
}

#[allow(dead_code)]
pub struct ServiceWrapper<'d> {
    node: Container<'d, Cli, Postgres>,
    url: String,
    pub client: Client,
}

impl<'d> ServiceWrapper<'d> {
    pub fn new() -> Self {
        info!("Creating service");

        let node = DOCKER.run(Postgres::default());

        let host = std::env::var("DOCKER_HOSTNAME").unwrap_or("localhost".to_owned());
        let port = node.get_host_port(5432).unwrap();
        let url = format!("postgres://postgres:postgres@{}:{}", host, port);

        let service = Service::new(url.clone(), Some(0)).unwrap();

        ServiceWrapper {
            node,
            url,
            client: service.client(),
        }
    }
}

pub fn assert_response(name: &'static str, mut response: LocalResponse) {
    let mut output = String::new();
    writeln!(output, "{}", response.status()).unwrap();

    let mut headers = response.headers().clone();
    headers.remove("X-Request-ID");
    for header in headers.iter() {
        writeln!(output, "{}", header).unwrap();
    }

    writeln!(output, "").unwrap();
    writeln!(
        output,
        "{}",
        response.body_string().unwrap_or("".to_owned())
    )
    .unwrap();
    assert_snapshot!(name, output);
}
