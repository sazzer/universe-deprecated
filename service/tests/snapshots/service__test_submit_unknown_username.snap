---
source: tests/service.rs
expression: output
---
200 OK
Content-Type: text/html; charset=utf-8
Server: Rocket

<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous" />

    <title>Universe</title>
  </head>
  <body>
    <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
  <a class="navbar-brand" href="/">Universe</a>
  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
    <span class="navbar-toggler-icon"></span>
  </button>

  <div class="collapse navbar-collapse" id="navbarSupportedContent">
    <ul class="navbar-nav ml-auto">
      <li class="nav-item">
        <a class="nav-link" href="/login">Login / Register</a>
      </li>
    </ul>
  </div>
</nav>


    <div class="container-fluid mt-3">
    
<div class="row">
  <div class="col-12 col-md-4 order-sm-1">
    
<h3>Register</h3>
<form action="/login/register" method="POST">
  <div class="form-group">
    <label for="login_username">Username</label>
    <input type="text"
           class="form-control "
           id="login_username"
           name="username"
           value="unknown"
           readonly />
    
  </div>
  <div class="form-group">
    <label for="login_email">Email Address</label>
    <input type="email"
           class="form-control "
           id="login_email"
           name="email"
           value=""
           aria-describedby="login_email_help"
           autofocus
           required />
    
    <small id="login_email_help" class="form-text text-muted">We'll never share your email with anyone else.</small>
  </div>
  <div class="form-group">
    <label for="login_name">Display Name</label>
    <input type="text"
           class="form-control "
           id="login_name"
           name="name"
           value=""
           required />
    
  </div>
  <div class="form-group">
    <label for="login_password">Password</label>
    <input type="password"
           class="form-control "
           id="login_password"
           name="password"
           required />
    
  </div>
  <div class="form-group">
    <label for="login_password2">Re-enter Password</label>
    <input type="password"
           class="form-control "
           id="login_password2"
           name="password2"
           required />
    
  </div>
  <button type="submit" class="btn btn-primary">Register</button>
</form>

  </div>
  <div class="col-12 col-md-8">
    <h3>The World of Sæmyyr</h3>
    <img src="/public/landing.jpg" width="100%" alt="The World of Sæmyyr"/>
  </div>
</div>

    </div>

    <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js" integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6" crossorigin="anonymous"></script>
  </body>
</html>


